File: VB/PrintingExample/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.Scheduling
Imports System.Collections.ObjectModel
Imports System.Data.Entity

Namespace PrintingExample
    <POCOViewModel> _
    Public Class MainViewModel
        Private receptionDeskContext As ReceptionDeskContext
        Public Overridable Property Doctors() As ObservableCollection(Of Doctor)
        Public Overridable Property Appointments() As ObservableCollection(Of MedicalAppointment)

        Protected Sub New()
            receptionDeskContext = New ReceptionDeskContext()
            receptionDeskContext.Doctors.Load()
            Doctors = receptionDeskContext.Doctors.Local
            receptionDeskContext.MedicalAppointments.Load()
            Appointments = receptionDeskContext.MedicalAppointments.Local
        End Sub
        Public Shared Function Create() As MainViewModel
            Return ViewModelSource.Create(Function() New MainViewModel())
        End Function
        Public Sub AppointmentsUpdated()
            receptionDeskContext.SaveChanges()
        End Sub

        #Region "#PrintScheduler"
        Public Sub PrintScheduler(ByVal scheduler As SchedulerControl)
            MyPrintHelper.PrintScheduler(scheduler)
        End Sub
        #End Region ' #PrintScheduler
    End Class
End Namespace

File: CS/PrintingExample/ViewModel/MainViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.Scheduling;
using System.Collections.ObjectModel;
using System.Data.Entity;

namespace PrintingExample {
    [POCOViewModel]
    public class MainViewModel {
        ReceptionDeskContext receptionDeskContext;
        public virtual ObservableCollection<Doctor> Doctors { get; set; }
        public virtual ObservableCollection<MedicalAppointment> Appointments { get; set; }

        protected MainViewModel() {
            receptionDeskContext = new ReceptionDeskContext();
            receptionDeskContext.Doctors.Load();
            Doctors = receptionDeskContext.Doctors.Local;
            receptionDeskContext.MedicalAppointments.Load();
            Appointments = receptionDeskContext.MedicalAppointments.Local;
        }
        public static MainViewModel Create() {
            return ViewModelSource.Create(() => new MainViewModel());
        }
        public void AppointmentsUpdated() {
            receptionDeskContext.SaveChanges();
        }

        #region #PrintScheduler
        public void PrintScheduler(SchedulerControl scheduler) {
            MyPrintHelper.PrintScheduler(scheduler);
        }
        #endregion #PrintScheduler
    }
}
